backend/controller/sql/queries.sql

```diff
@@ -731,3 +731,35 @@ UPDATE topic_subscriptions
 SET state = 'idle'
 WHERE name = @name::TEXT
       AND module_id = (SELECT id FROM module);
+
+-- name: GetSubscription :one
+WITH module AS (
+  SELECT id
+  FROM modules
+  WHERE name = $2::TEXT
+)
+SELECT *
+FROM topic_subscriptions
+WHERE name = $1::TEXT
+      AND module_id = (SELECT id FROM module);
+
+-- name: SetSubscriptionCursor :exec
+WITH event AS (
+  SELECT id, created_at, key, topic_id, payload
+  FROM topic_events
+  WHERE "key" = $2::topic_event_key
+)
+UPDATE topic_subscriptions
+SET state = 'idle',
```

Comment:
I think this could cause a race condition, maybe?

- Subscription is behind and currently consuming an event
- Reset subscription command comes in sets the state to `idle` and updates the cursor to head
- 2 new events are posted to the topic
- New event starts being consumed by the subscription
- Original event (from before the reset command) finishes being processed, sets the state to `idle` even though we are processing the other event still
- Then pub sub triggers the next event to be processed (without knowing the other event is being processed still)

Reply:
ah interesting, do you think the solution would be to only update the cursor and not the state? so that if there are any outstanding async calls executing, they'd have to complete before any new ones are kicked off

Reply:
I honestly don't know, possibly! I added it as an issue because I'm not sure whether its a priority but didn't want it to be lost: https://github.com/TBD54566975/ftl/issues/2261